extension/test/e2e/affirm.spec.js

```diff
@@ -0,0 +1,267 @@
+const RedirectPaymentFormPage = require('./pageObjects/RedirectPaymentFormPage')
+const AffirmPage = require('./pageObjects/AffirmPage')
+const {
+  CTP_INTERACTION_TYPE_MANUAL_CAPTURE,
+} = require('../../src/config/constants')
+
+// Flow description: https://docs.adyen.com/payment-methods/klarna/web-component#page-introduction
+describe('::affirmPayment::', () => {
+  let browser
+  let ctpClient
+  const adyenMerchantAccount = config.getAllAdyenMerchantAccounts()[0]
+  const ctpProjectKey = config.getAllCtpProjectKeys()[0]
+
+  beforeEach(async () => {
+    routes['/make-payment-form'] = async (request, response) => {
+      serveFile(
+        './test/e2e/fixtures/affirm-make-payment-form.html',
+        request,
+        response
+      )
+    }
+    routes['/redirect-payment-form'] = async (request, response) => {
+      serveFile(
+        './test/e2e/fixtures/redirect-payment-form.html',
+        request,
+        response
+      )
+    }
+    routes['/return-url'] = async (request, response) =>
+      httpUtils.sendResponse({
+        response,
+        headers: {
+          'Content-Type': 'text/html',
+        },
+        data:
+          '<!DOCTYPE html><html><head></head>' +
+          '<body id=redirect-response>' +
+          'This is a return page to show users after they finish the payment' +
+          '</body></html>',
+      })
+
+    const ctpConfig = config.getCtpConfig(ctpProjectKey)
+    ctpClient = ctpClientBuilder.get(ctpConfig)
+    browser = await initPuppeteerBrowser()
+  })
+
+  afterEach(async () => {
+    await browser.close()
+  })
+
+  it(
+    'when payment method is affirm and process is done correctly, ' +
+      'then it should successfully finish the payment',
+    async () => {
+      const baseUrl = config.getModuleConfig().apiExtensionBaseUrl
+      const clientKey = config.getAdyenConfig(adyenMerchantAccount).clientKey
+      try {
+        const payment = await createPayment(
+          ctpClient,
+          adyenMerchantAccount,
+          ctpProjectKey,
+          'USD'
+        )
+        logger.debug('affirm::payment:', JSON.stringify(payment))
+        const browserTab = await browser.newPage()
+
+        const paymentAfterMakePayment = await makePayment({
+          browserTab,
+          baseUrl,
+          payment,
+          clientKey,
+        })
+        logger.debug(
+          'affirm::paymentAfterMakePayment:',
+          JSON.stringify(paymentAfterMakePayment)
+        )
+        const paymentAfterHandleRedirect = await handleRedirect({
+          browserTab,
+          baseUrl,
+          payment: paymentAfterMakePayment,
+        })
+        logger.debug(
+          'affirm::paymentAfterHandleRedirect:',
+          JSON.stringify(paymentAfterHandleRedirect)
+        )
+        assertPayment(paymentAfterHandleRedirect)
+
+        // Capture the payment
+        const paymentAfterCapture = await capturePayment({
```

Comment:
I guess you don't need to do `capturePayment` in e2e test. It is done without UI anyway in the backend. If you have IT for that, it's enough.

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/769/commits/d2affaffcf7ca42da069f927df40066e0278e4ba